spec/requests/omniauth_callbacks_controller_spec.rb

```diff
@@ -33,6 +33,39 @@
       expect(Users::OmniauthCallbacksController.find_authenticator("twitter"))
         .not_to eq(nil)
     end
+
+    context "with a plugin-contributed auth provider" do
+      before do
+        @provider = Plugin::AuthProvider.new
+
+        @provider.authenticator = Auth::OpenIdAuthenticator.new('ubuntu', 'https://login.ubuntu.com', trusted: true)
+        @provider.enabled_setting = "ubuntu_login_enabled"
+
+        Discourse.stubs(:auth_providers).returns [ @provider ]
+      end
+
+      it "finds an authenticator when enabled" do
+        SiteSetting.stubs(:ubuntu_login_enabled).returns(true)
```

Comment:
Instead of stubbing SiteSetting, you can say `SiteSetting.ubuntu_login_enabled = true`

Reply:
`ubuntu_login_enabled` isn't a real Site Setting (hypothetically it is one that would be added by a plugin)

I can programmatically define a new SiteSetting at the beginning of the test by doing
```
SiteSetting.setting('ubuntu_login_enabled', true)
```
but I'm unsure how to properly clean that up at the end of the test.